### PR TITLE
Extend info about `group` with an antipattern section

### PR DESCRIPTION
**Pull request by ppcano**
Discourage the usage of `group` in some cases and provide alternatives.

### CODE REVIEW

File: src/data/markdown/docs/01 guides/02 Using k6/08 Tags and Groups.md [hunk 1]
```diff
@@ -43,6 +44,37 @@ execution of the supplied `group()` function also emits a `group_duration`
...
+
+</CodeGroup>
+
+Groups were designed to help you visualize the various parts of a large test. 
+If your code looks like the example above, consider the following alternatives:
+
+- For dynamic URLs, use the [URL grouping feature](/using-k6/http-requests#url-grouping).
+- To provide a meaningful name to your request, set the value of [tags.name](/using-k6/http-requests#http-request-tags).
+- To reuse common logic or organize your code better, create a [local Javascript module](/using-k6/modules#local-filesystem-modules) and import it into the test script.
+
```
Comment (na--) [inline src/data/markdown/docs/01 guides/02 Using k6/08 Tags and Groups.md:74]: Maybe link to the scenarios page here as well?

File: src/data/markdown/docs/01 guides/02 Using k6/08 Tags and Groups.md [hunk 2]
```diff
@@ -24,13 +24,14 @@ allowing you the BDD-style of testing.
...
-      // perform login
-    });
+  group('add several products to the shopping cart', function () {
+    // ...
+  });
+  group('go to login and authenticate', function () {
+    // ...
+  });
+  group('checkout process', function () {
+    // ...
```
Comment (MStoykov) [inline src/data/markdown/docs/01 guides/02 Using k6/08 Tags and Groups.md:34]: I am for at least one example with nested groups, but it should probably be not an anti-pattern one ;)

File: src/data/markdown/docs/01 guides/02 Using k6/08 Tags and Groups.md (hunk 1 above)
Comment (ppcano) [reply src/data/markdown/docs/01 guides/02 Using k6/08 Tags and Groups.md:74]: agree

File: src/data/markdown/docs/01 guides/02 Using k6/08 Tags and Groups.md (hunk 2 above)
Comment (ppcano) [reply src/data/markdown/docs/01 guides/02 Using k6/08 Tags and Groups.md:34]: I intentionally removed it to simplify the example. 

Not saying that is the best option.

It is mentioned in the description.

File: src/data/markdown/docs/01 guides/02 Using k6/08 Tags and Groups.md (hunk 1 above)
Comment (ppcano) [reply src/data/markdown/docs/01 guides/02 Using k6/08 Tags and Groups.md]: solved by https://github.com/loadimpact/k6-docs/pull/210/commits/aa00ae948aa7739e50c4b726e156ec78a0c3ec18

**Review by nicolek6 — CHANGES_REQUESTED** [review]
I agree that it doesn't make sense to have a group for just one request. But I think we should recommend using groups for multiple requests that are still a result of only one page/action. For those spanning more than one action, I think we should recommend functions rather than groups.

File: src/data/markdown/docs/01 guides/02 Using k6/08 Tags and Groups.md [hunk 3]
```diff
@@ -43,6 +44,37 @@ execution of the supplied `group()` function also emits a `group_duration`
...
+
+</CodeGroup>
+
+Groups were designed to help you visualize the various parts of a large test. 
+If your code looks like the example above, consider the following alternatives:
+
+- For dynamic URLs, use the [URL grouping feature](/using-k6/http-requests#url-grouping).
+- To provide a meaningful name to your request, set the value of [tags.name](/using-k6/http-requests#http-request-tags).
+- To reuse common logic or organize your code better, create a [local Javascript module](/using-k6/modules#local-filesystem-modules) and import it into the test script.
+- If you need to model advanced user patterns, check out [Scenarios](/using-k6/scenarios).
```
[the remark this answers is not in the document]
Comment (nicolek6) [reply src/data/markdown/docs/01 guides/02 Using k6/08 Tags and Groups.md:74]: Oops, just saw the local JavaScript module part. Takes care of my function suggestion, but maybe comments would be worth mentioning? Up to you.

**Review by simskij — CHANGES_REQUESTED** [review]

File: src/data/markdown/docs/01 guides/02 Using k6/08 Tags and Groups.md [hunk 4]
```diff
@@ -43,6 +44,37 @@ execution of the supplied `group()` function also emits a `group_duration`
 function. This, combined with the metric tags described below, enables very flexible performance
 monitoring of different groups in your test suite.
 
+**Group anti-pattern**
```
[the remark this answers is not in the document]
Comment (ppcano) [reply src/data/markdown/docs/01 guides/02 Using k6/08 Tags and Groups.md]: https://github.com/loadimpact/k6-docs/pull/210/commits/4cb554c931568a543d34e4220e53a5d5cf36bf6d

**Comment by ppcano** [comment]
@nicolek6 Please, let me know if the latest changes address your points. You can read it https://mdr-ci.staging.k6.io/docs/refs/pull/210/merge/using-k6/tags-and-groups

**Comment by simskij** [comment]
Nothing more from me. I'll leave the approval to @MStoykov or @nicolek6 as they both have unresolved discussions in their reviews. 👍🏼

**Review by nicolek6 — APPROVED** [review]
I really like the new changes, Pepe. I think it reads much better!

**Pull request merged**